packages/aws-cdk-lib/custom-resources/test/custom-resource-config/custom-resource-config.test.ts

```diff
@@ -310,3 +311,47 @@ describe('when custom resource logGroup removalPolicy is Retain', () => {
     });
   });
 });
+
+describe('when custom resource lambda runtime is modified by addLambdaRuntime', () => {
+  test('addLambdaRuntime modifies custom resource lambda runtime to python3.12', () => {
+    // GIVEN
+    const customResourceRuntime = lambda.Runtime.PYTHON_3_12;
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app);
+    const websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 'BucketDeployment', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+    });
+
+    // WHEN
+    CustomResourceConfig.of(app).addLambdaRuntime(customResourceRuntime);
+
+    // THEN
+    const template = Template.fromStack(stack);
+    template.hasResourceProperties('AWS::Lambda::Function', {
+      Runtime: customResourceRuntime.toString(),
+    });
+  });
+
+  test('addLambdaRuntime does not modify custom resource lambda runtime in a different runtime family', () => {
+    // GIVEN
+    const customResourceRuntime = lambda.Runtime.NODEJS_20_X;
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app);
```

Comment:
Please also add a test that modifies only a custom resource handler but ignores a standalone lambda